### PR TITLE
Add legend

### LINKED ISSUE #339: Add a colour legend (if number of clusters is not too big).

### PR DESCRIPTION
**Pull request by federikovi**
Added legend

Fixes #339

---

Checklist
---

- [ ] Added link to Github issue or Trello card
- [ ] Added tests

### CODE REVIEW

File: wellcomeml/viz/visualize_clusters.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    p = figure(title="Cluster visualization", toolbar_location="above",
+               plot_width=plot_width, plot_height=plot_height,
+               tools=tools, tooltips=tooltips,
+               background_fill_color=well_background,
+               sizing_mode='scale_width')
+
+    R = []
+    for x in clusters_uniq:
+        df = data[data['cluster_id'] == str(x)]
+        r = p.circle(x="X", y="Y", color="colors", source=df)
```
Comment (aCampello) [inline wellcomeml/viz/visualize_clusters.py:64]: what happened to the radius and alpha options?

Comment (aCampello) [reply wellcomeml/viz/visualize_clusters.py:65]: Can we add: 
```radius=radius, fill_alpha=alpha``` to the p.circle? Looks nicer locally to me :)

Comment (aCampello) [reply wellcomeml/viz/visualize_clusters.py:65]: Also `alpha=0.5, radius=0.05` look nice on my machine :D

**Review by aCampello — CHANGES_REQUESTED** [review]
Thanks for this - Legend looks good. Can we add in the fill_alpha and radius again?

File: wellcomeml/viz/visualize_clusters.py (hunk 1 above)
Comment (federikovi) [reply wellcomeml/viz/visualize_clusters.py:65]: you're right! I set that value as default

**Review by aCampello — APPROVED** [review]
Go for it!

**Pull request merged**